apps/web/src/pages/VerifyEmail.tsx

```diff
@@ -1,14 +1,101 @@
-import { EmailVerification } from "@litespace/ui/EmailVerification";
-import { Route } from "@/types/routes";
-import React, { useCallback } from "react";
+import Aside from "@/components/Auth/Aside";
+import Header from "@/components/Auth/Header";
+import {
+  EmailVerification,
+  VerificationState,
+} from "@/components/Auth/EmailVerification";
+import React, { useCallback, useEffect, useState } from "react";
+import { useSendVerifyEmail, useVerifyEmail } from "@litespace/headless/auth";
+import { useNavigate, useSearchParams } from "react-router-dom";
+import { useFormatMessage } from "@litespace/ui/hooks/intl";
+import { useToast } from "@litespace/ui/Toast";
+import { QueryKey } from "@litespace/headless/constants";
+import { useInvalidateQuery } from "@litespace/headless/query";
+import { VERIFY_EMAIL_CALLBACK_URL, Route } from "@/types/routes";
 
 const VerifyEmail: React.FC = () => {
-  const onVerification = useCallback(() => {
-    // todo: refersh user data
-  }, []);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [verificationState, setVerificationState] =
+    useState<VerificationState>("loading");
+  const navigate = useNavigate();
+  const intl = useFormatMessage();
+  const toast = useToast();
+  const [token, setToken] = useState<string | null>(null);
+  const invalidateQuery = useInvalidateQuery();
+
+  const onVerifySuccess = useCallback(() => {
+    setVerificationState("success");
+    invalidateQuery([QueryKey.FindCurrentUser]);
+  }, [invalidateQuery]);
+
+  const goDashboard = useCallback(() => {
+    navigate(Route.Root);
+  }, [navigate]);
+
+  const onVerifyError = useCallback(() => {
+    toast.error({ title: intl("page.verify.email.failure") });
+    setVerificationState("error");
+  }, [toast, intl]);
+
+  const verifyMutation = useVerifyEmail({
+    onSuccess: onVerifySuccess,
+    onError: onVerifyError,
+  });
+
+  const onResendSuccess = useCallback(() => {
+    toast.success({
+      title: intl("page.check.email.resend.success"),
+    });
+  }, [toast, intl]);
+
+  const onResendError = useCallback(() => {
+    toast.error({
+      title: intl("page.check.email.resend.error"),
+    });
+  }, [toast, intl]);
+
+  const resendMutation = useSendVerifyEmail({
+    onSuccess: onResendSuccess,
+    onError: onResendError,
+  });
+
+  const resendEmail = useCallback(() => {
+    resendMutation.mutate(VERIFY_EMAIL_CALLBACK_URL);
+    setVerificationState("resend");
+  }, [resendMutation]);
```

Comment:
The resend button should be in the loading & disabled state while re-sending the email.